pilot/pkg/trustbundle/trustbundle.go

```diff
@@ -151,53 +162,77 @@ func (tb *TrustBundle) mergeInternal() {
 }
 
 // UpdateTrustAnchor: External Function to merge a TrustAnchor config with the existing TrustBundle
-func (tb *TrustBundle) UpdateTrustAnchor(anchorConfig *TrustAnchorUpdate) error {
+func (tb *TrustBundle) updateTrustAnchorInternal(anchorConfig *TrustAnchorUpdate) (bool, error) {
 	var ok bool
 	var err error
 
+	tb.mutex.Lock()
```

Comment:
@howardjohn - there are two go-routines that call this function,  hence grabbing the write locks. The scope of this function is large now and could get larger (reading/writing into other parts of the trustbundle). As I discussed earlier - I think its better this function is only ever accessed from one go-routine, given its scope. This would require all meshConfig updates to be written into a channel and read by the same thread that processes remote Trust Anchor updates. Does this sound good?